homeassistant/components/starlink/coordinator.py

```diff
@@ -80,3 +86,45 @@ async def async_reboot_starlink(self) -> None:
                 await self.hass.async_add_executor_job(reboot, self.channel_context)
             except GrpcError as exc:
                 raise HomeAssistantError from exc
+
+    async def async_set_sleep_schedule_enabled(self, sleep_schedule: bool) -> None:
+        """Set whether Starlink system uses the configured sleep schedule."""
+        async with asyncio.timeout(4):
```

Comment:
Instead of using magic number 4 everywhere, we should make a constant `DEFAULT_TIMEOUT` or something, but that could happen in a separate PR.
Also, it doesn't seem like anything is catching the timeout errors, which means that users will see stack traces if it happens. The timeout errors should be caught and nicely logged IMO. Again, that can happen in a separate PR since it also affects already existing code.